app/src/main/java/at/aau/edu/wizards/gameboard/GameboardFragment.kt

```diff
@@ -0,0 +1,68 @@
+package at.aau.edu.wizards.gameboard
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+
+import at.aau.edu.wizards.R
+
+
+class GameboardFragment : Fragment() {
+
+    private lateinit var adapter: GameboardAdapter
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var cardsArrayList: ArrayList<Cards>
+
+    lateinit var imageId : Array<Int>
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_gameboard, container, false)
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        dataInitialize()
+
+        val layoutManager = GridLayoutManager(context, 1,
+                LinearLayoutManager.HORIZONTAL, true)
+        recyclerView = view.findViewById(R.id.gameboard_recycler_view)
```

Comment:
`lateinit` allows assignment only once. If you leave the application it will destroy the view, on re-entering it will call again `onCreateView`, then `onViewCreated`. Here you would assign the value a second time and the app will crash.
I almost never use `lateinit`, it's error prone if done wrong.

Please use ViewBinding, basically you can look that up inside `SampleFragment` - just do the same with different names.